File: Assets/Typescript~/src/Shared/Projectile/ProjectileSharedImpl.ts
import { Projectile } from "./Projectile";

export class ProjectileSharedImpl {
	/**
	 *
	 * @param projectile
	 * @param hitPoint
	 * @param hitVelocity
	 * @param collider
	 * @returns true if collision is ignored.
	 */
	public static ShouldIgnoreCollision(
		projectile: Projectile,
		hitPoint: Vector3,
		velocity: Vector3,
		collider: Collider,
	): boolean {
		return false;
	}
}

File: Assets/Bundles/Shared/Resources/TS/Projectile/ProjectileSharedImpl.lua
-- Compiled with unity-ts v2.1.0-75
local ProjectileSharedImpl
do
	ProjectileSharedImpl = setmetatable({}, {
		__tostring = function()
			return "ProjectileSharedImpl"
		end,
	})
	ProjectileSharedImpl.__index = ProjectileSharedImpl
	function ProjectileSharedImpl.new(...)
		local self = setmetatable({}, ProjectileSharedImpl)
		return self:constructor(...) or self
	end
	function ProjectileSharedImpl:constructor()
	end
	function ProjectileSharedImpl:ShouldIgnoreCollision(projectile, hitPoint, velocity, collider)
		return false
	end
end
return {
	ProjectileSharedImpl = ProjectileSharedImpl,
}
-- ----------------------------------
-- ----------------------------------
